### PR TITLE
Promote pipeline: Correct setup of repository in build.gradle files (#1467)

### PR DESCRIPTION
**Pull request by radtriste**
backport from: https://github.com/kiegroup/optaplanner/pull/1467

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] native</b>
</details>

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.promote [hunk 1]
```diff
@@ -557,8 +557,8 @@ void runMavenDeploy(MavenCommand mvnCmd) {
 void removeJbossNexusFromMavenAndGradle() {
     sh "sed -i \':a;N;\$!ba;s/\\n *<repositories>.*<\\/repositories>//g\' */pom.xml"
     sh "sed -i \':a;N;\$!ba;s/\\n *<repositories>.*<\\/repositories>//g\' pom.xml"
```
Comment (rsynek) [inline .ci/jenkins/Jenkinsfile.promote:559]: With the change of the `optaplanner-quickstarts` structure, I think also these two lines need to be replaced by something similar:
```
find . -name pom.xml -print0 | xargs -0 sed -i \':a;N;\$!ba;s/\\n *<repositories>.*<\\/repositories>//g\'
```

Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.promote]: updated

**Review by rsynek — APPROVED** [review]
Thanks for the fix!

**Review by rsynek — APPROVED** [review]

**Comment by rsynek** [comment]
Merging as these files are not covered by the CI.

**Pull request merged**